updater/lib/dependabot/dependency_group_engine.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require "dependabot/dependency_group"
+
+# This class implements our strategy for keeping track of and matching dependency
+# groups that are defined by users in their dependabot config file.
+#
+# Each UpdateJob registers its own DependencyGroupEngine which calculates
+# the grouped and ungrouped dependencies for a DependencySnapshot
+#
+# This is a static class tied to the lifecycle of a Job
+# Its methods should only be called with Dependabot::DependencyGroupEngine
+#
+# Groups are only calculated once after the Job has registered its dependencies
+# All allowed dependencies should be passed in to the calculate_dependency_groups! method
```

Comment:
Thanks, those bullet points read easier to me too.

For "Its methods should only be called with Dependabot::DependencyGroupEngine" my original intention was to signal that this class follows a singleton pattern (for now) and can only be accessed statically.

I think it's fine to leave that line out though since I plan on refactoring this class to one that can be instantiated.